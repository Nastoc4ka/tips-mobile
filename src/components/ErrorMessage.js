import React from 'react';
import { View, Text } from 'react-native';
import { errors } from '../styles';

const ErrorMessage = ({message, styles = []}) => {
    return (
        <View style={[errors.wrapper, {...styles.wrapper}]}>
            <Text numberOfLines={5} style={[errors.text, {...styles.text}]}>{message}</Text>
        </View>
    )
};

export default ErrorMessage;