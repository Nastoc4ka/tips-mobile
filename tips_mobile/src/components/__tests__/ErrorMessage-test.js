import React from 'react';
import renderer from 'react-test-renderer';
import ErrorMessage from '../ErrorMessage';
import {Platform} from 'react-native';

describe('Tests ErrorMessage', () => {
    describe("IOS", () => {

        beforeAll(() => {
            Platform.OS = 'ios';
        });

        test('component render properly' ,() => {
            const actual = renderer.create(<ErrorMessage message={'some small error'}/>);

            expect(actual.toJSON()).toMatchSnapshot();
        });
    });

    describe("Android", () => {

        beforeAll(() => {
            Platform.OS = 'android';
        });

        test('component render properly' ,() => {
            const actual = renderer.create(<ErrorMessage message={'some small error'}/>);

            expect(actual.toJSON()).toMatchSnapshot();
        });
    });
});
