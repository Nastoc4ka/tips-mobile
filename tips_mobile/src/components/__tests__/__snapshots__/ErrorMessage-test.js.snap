// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests ErrorMessage Android component render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "marginRight": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "red",
        "fontSize": 12,
      }
    }
  >
    some small error
  </Text>
</View>
`;

exports[`Tests ErrorMessage IOS component render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "marginRight": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "red",
        "fontSize": 12,
      }
    }
  >
    some small error
  </Text>
</View>
`;

